### PR TITLE
feat: Enforce usage of parent functions for main viewset

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

One can accidentally override certain functions that are designed to protect the API.

## Changes

* Enforce usage of these functions via a subclass init which ensures it hasn't been tampered with
* Adds extra `safely_` methods for overriding things the right way and `dangerously_` versions that completely override but hopefully making this extra clear that it is dangerous.
* Also I believe fixes a bug that meant the `TaggedItemViewSetMixin` method was being ignored

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Review by webjunkie — COMMENTED** [review]
Good ideas! Had a look and it looks alright as far as I can see.

**Review by pauldambra — APPROVED** [review]
👍 clicked around and things seem to work
👍 the naming is very bueno for making people think about what they're doing

File: posthog/api/organization.py [hunk 1]
```diff
@@ -148,21 +148,23 @@ class OrganizationViewSet(TeamAndOrgViewSetMixin, viewsets.ModelViewSet):
     ordering = "-created_by"
 
     def dangerously_get_permissions(self):
-        permission_classes = self.permission_classes
-
-        # When listing there is no individual object to check for
         if self.action == "list":
-            permission_classes = [permissions.IsAuthenticated, APIScopePermission]
-        elif self.action == "create":
+            return [permission() for permission in [permissions.IsAuthenticated, APIScopePermission]]
```
Comment (webjunkie) [inline posthog/api/organization.py:152]:
```suggestion
            return [permissions.IsAuthenticated(), APIScopePermission()]
```

Wouldn't that be shorter?

**Review by webjunkie — APPROVED** [review]
👍🏻

File: posthog/api/organization.py (hunk 1 above)
Comment (benjackwhite) [reply posthog/api/organization.py:152]: _yes_ but I was aiming for consistency over brevity. Trying to stick to the other methods and how DRF does things

**Pull request merged**